src/transform/dae_finish.jl

```diff
@@ -262,6 +262,19 @@ end
 
 const VectorViewType = SubArray{Float64, 1, Vector{Float64}, Tuple{UnitRange{Int}}, true}
 
+function cache_dae_ci!(old_ci, src, debuginfo, owner)
+    daef_ci = CC.engine_reserve(old_ci.def, owner)
+    ccall(:jl_fill_codeinst,  Cvoid, (Any, Any, Any, Any, Int32, UInt, UInt, UInt32, Any, Any, Any),
+        daef_ci, Tuple{}, Union{}, nothing, Int32(0),
+        UInt(1)#=ci.min_world=#, old_ci.max_world,
+        old_ci.ipo_purity_bits, nothing, nothing, CC.empty_edges)
+    ccall(:jl_update_codeinst, Cvoid, (Any, Any, Int32, UInt, UInt, UInt32, Any, UInt8, Any, Any),
+        daef_ci, src, Int32(0), UInt(1)#=ci.min_world=#, old_ci.max_world, old_ci.ipo_purity_bits,
+        nothing, 0x0, debuginfo, CC.empty_edges)
+    ccall(:jl_mi_cache_insert, Cvoid, (Any, Any), old_ci.def, daef_ci)
```

Comment:
Don't use these as ccalls as the function is internal and likely to change often. You also need to close your engine reservation, but you probably shouldn't be making a reservation at all as this seems probably too late to acquire one. If you know that recursion is structurally unsupported by this package, just use the CodeInstance constructor?